### PR TITLE
MM-39364: Inline task description in RHS

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
Add collapsible (default expanded) task descriptions to the RHS.

https://user-images.githubusercontent.com/3924815/142251913-7bd61711-2a43-498b-bcba-f89457ac316d.mp4

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39364

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/checklist_item.tsx [hunk 1]
```diff
@@ -224,6 +235,10 @@ const ChecklistItemDescription = styled.div`
 
         white-space: pre-wrap;
     }
+    height: ${({height}) => height};
+
+    transition: height 0.2s ease-in-out;
```
Comment (cpoile) [inline webapp/src/components/checklist_item.tsx:240]: slick!

**Review by cpoile — APPROVED** [review]
This is awesome, nice work @agarciamontoro! Nice use of a ref and scrollHeight.

File: webapp/src/components/checklist_item.tsx (hunk 1 above)
Comment (agarciamontoro) [reply webapp/src/components/checklist_item.tsx:240]: Thanks! If only I could avoid the triple `height`... :joy:

**Review by abhijit-singh — APPROVED** [review]
Excellent @agarciamontoro!

**Review by crspeller — APPROVED** [review]
Nice!

**Pull request merged**